sdf/final-project/technical.md

```diff
@@ -6,102 +6,138 @@ This is intended to be filled out by staff members of DPI Tech Apprenticeship pr
+- [ ] **Comment Quality**: Does the code include inline comments that explain "why" behind non-obvious logic? Over-commenting should be avoided; code should be self-explanatory wherever possible. 
+- [ ] **Minimal Unused Code**: Unused code should be deleted (not commented out).
 
 ### Notes:
 
-## Frontend
-- [ ] **Semantic HTML**: Is the project making effective use of semantic HTML elements to structure the content, ensuring that it's both accessible and SEO-friendly?
-- [ ] **Styling**: Does the frontend employ CSS or CSS frameworks (like Bootstrap) for styling?
-- [ ] **Template Usage for Reusability**: Are partial templates used to promote code reusability and maintainability for common UI components (e.g., navigation bars, forms, lists)?
-- [ ] **Client-Side Interactivity**: Is JavaScript or JavaScript frameworks/libraries (e.g., jQuery) utilized to enhance the UI with dynamic content and interactive elements, reducing unnecessary page reloads?
-- [ ] **Rails View Helpers**: Does the project leverage Rails view helpers (link_to, form_with, etc.) to streamline the generation of view code and enforce best practices in form and link creation?
-
-### Bonus (max 2pts)
-- [ ] **Responsive Design**: It looks and works great on all different devices (seamless user experience across desktop, tablet, and mobile)
-- [ ] **Accessibility Considerations**: Are accessibility features (such as alt tags for images, ARIA roles) implemented to support users with disabilities and to comply with web accessibility standards?
-- [ ] **Enhanced Navigation and Feedback**: Are breadcrumbs used to enhance site navigation? Are styled flashes or toasts implemented to provide clear, user-friendly feedback? (in a partial)
+## Patterns of Enterprise Applications (max: 8 points)
+- [ ] **Domain Driven Design**: Does the application follow domain-driven design principles, with clear separation of concerns and a focus on the core domain logic?
+- [ ] **Advanced Data Modeling**: Has the application utilized ActiveRecord callbacks for model lifecycle management?
+- [ ] **Component-Based View Templates**: Does the application use component-based view templates (partials) to promote reusability and maintainability?
+- [ ] **Modules**: Does the application use modules (concerns) to encapsulate related functionality and promote code organization?
+- [ ] **Service Objects**: Does the application abstract logic into service objects when appropriate?
+- [ ] **Polymorphism**: Does the application use polymorphism to create flexible and maintainable code?
+- [ ] **Overall Separation of Concerns**: Are the concerns of the application (e.g., data access, business logic, presentation) separated effectively, with each layer focused on its specific responsibilities?
+- [ ] **Overall DRY Principle**: Does the application follow the DRY (Don't Repeat Yourself) principle, with code reuse and minimal redundancy?
 
+### Notes:
 
-### Score (/5):
+## Frontend (max: 10 points)
```

Comment:
maybe we want to incorporate some of the tips in the design resources section https://learn.firstdraft.com/lessons/143-design-resources

specifically these 
https://practicaltypography.com/typography-in-ten-minutes.html
https://www.learnui.design/blog/7-rules-for-creating-gorgeous-ui-part-1.html
https://www.learnui.design/blog/7-rules-for-creating-gorgeous-ui-part-2.html#rule-4-learn-the-methods-of-overlaying-text-on-images


some key rules

Rule 3: Double your whitespace

make sure text readable (eg don't use white text on bright pink)

Reply:
I gave it a shot, see changes